redis/src/main/scala/zio/redis/api/Hashes.scala

```diff
@@ -110,13 +154,19 @@ trait Hashes {
    * @param count approximate number of elements to return (see https://redis.io/commands/scan#the-count-option)
    * @return pair containing the next cursor and list of elements scanned
    */
-  final def hScan(
-    key: String,
+  final def hScan[K: Schema, F: Schema, V: Schema](
+    key: K,
     cursor: Long,
     pattern: Option[String] = None,
     count: Option[Count] = None
-  ): ZIO[RedisExecutor, RedisError, (Long, Chunk[String])] =
-    HScan.run((key, cursor, pattern.map(Pattern), count))
+  ): ZIO[RedisExecutor, RedisError, (Long, Map[F, V])] = {
```

Comment:
@mijicd is it ok to change output contract from chank to map, it's a big difference?
I'am working now on sorted sets and i want to change outputs contracts there too.

Reply:
https://github.com/anatolysergeev/zio-redis/pull/8/files
I've written TODOs every where i want to change output contract and some where i'm think to split methods on two in api (because depending on flag output is different), what do you think about that?

Reply:
Sure.

Reply:
I think we need to change the output to at least chunk of tuples `Chunk[(F, V)]`. Maybe changing it to `Map` directly is too much but in my opinion, chunk of Strings is too low level and the user needs to 

1. Understand the structure of the output 
2. Write code to further work with the output

So I would agree that change the output to `Map` is maybe too high level but as I said we should at least change it to `Chunk[(F,  K)]`

Reply:
I agree with you, that we should to do an opportunity to work with the bite API. But may be for the hashes it would be reasonable to do `Map`, in sorted set i can't do that. If we want to save a similarity between `SCAN` commands lets do `Chunk[(F,K)]` if not `Map` is fine too.
![image](https://user-images.githubusercontent.com/8873163/113748171-d98b7b80-9710-11eb-9d8e-fa2db66bb6d0.png)